core/src/main/java/aima/core/search/basic/uninformed/BidirectionalSearch.java

```diff
@@ -38,22 +38,37 @@
     @Override
 
     public BidirectionalActions<A> apply(Problem<A, S> originalProblem, Problem<A, S> reverseProblem) {
-        Node<A, S> node = newRootNode(originalProblem.initialState(), 0);
+        Node<A, S> node = newRootNode(originalProblem.initialState(), 0); //node -> starting point 
+        //if starting node is the goal node then return bidirectionalAction 
+        //which is the solution of the problem (total cost -> zero)
+        //First terminating condition
         if (originalProblem.isGoalState(node.state())) {
             this.previousMeetingOfTwoFrontiers = node;
             this.nextNodeToBeEvaluated = null;
             bidirectionalActions = new BasicBidirectionalActions<>(fromInitialStatePart(), fromGoalStatePart());
             return bidirectionalActions;
         }
-        Node<A, S> revNode = newRootNode(reverseProblem.initialState(), 0);
+        
+        Node<A, S> revNode = newRootNode(reverseProblem.initialState(), 0);//revNode -> goal node
+        //front -> priority queue which has node to be explored when traversing from starting node
```

Comment:
don't include variable names in comments. when names change, your comment is obsolete